hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/KeyOutputStream.java

```diff
@@ -303,7 +303,7 @@ private int writeToOutputStream(BlockOutputStreamEntry current,
       if (retry) {
         current.writeOnRetry(len);
       } else {
-        waitForRetryHandling(current);
+        current.waitForRetryHandling(retryHandlingCondition);
```

Comment:
we don't need the write lock here because it is already synchronized?

Reply:
Yeah, because this method is already inside the writeLock.